### PR TITLE
Add post about nifs

### PR DESCRIPTION
**Pull request by ConnorRigby**
This ended up being a little longer than I wanted, but here's a real world nif example.

### CODE REVIEW

**Comment by fhunleth** [comment]
Awesome!! 

I made a quick pass at some edits at https://github.com/fhunleth/embedded-elixir/commits/nifs. I gave you permission to the repo to make collaboration easier.

I need to go do Christmas stuff with the family, but I think this is getting close. I did want to update the section on the `CROSSCOMPILE` and `CC` variables in the `Makefile`. Nerves overrides `CC`. The `CROSSCOMPILE` variable is more of a convention for non-Nerves users, but I think that I messed up the Makefile code a long time ago since I think that `CC` is always defined. It should be an if `CROSSCOMPILE` is defined, then override `CC`. I think that the whole `CROSSCOMPILE` stuff should be deleted, but I'm not at a good place to test.

File: content/post/2017-12-23-using-nifs-with-nerves.md [hunk 1]
```diff
@@ -114,12 +114,12 @@ end
 
 @compile {:inline, [time_unit_to_seconds: 2]}
 defp time_unit_to_seconds(_, "never"), do: 0
-defp time_unit_to_seconds(repeat, "minutely"), do: 60 * repeat
-defp time_unit_to_seconds(repeat, "hourly"), do: 60 * 60 * repeat
-defp time_unit_to_seconds(repeat, "daily"), do: 60 * 60 * 24 * repeat
-defp time_unit_to_seconds(repeat, "weekly"), do: 60 * 60 * 24 * 7 * repeat
-defp time_unit_to_seconds(repeat, "monthly"), do: 60 * 60 * 24 * 30 * repeat
-defp time_unit_to_seconds(repeat, "yearly"), do: 60 * 60 * 24 * 365 * repeat
+defp time_unit_to_seconds(repeat, "minutely"), do: 60
```
Comment (fhunleth) [inline content/post/2017-12-23-using-nifs-with-nerves.md]: Did you mean to trim off the ` * repeat`?

**Comment by ConnorRigby** [comment]
Yes I did. Was bad math.

**Comment by ConnorRigby** [comment]
Guess that arg isnt needed either then..

**Comment by fhunleth** [comment]
Could you also merge in my edits from https://github.com/fhunleth/embedded-elixir/commit/3cfb9be5a2e516d5d494a80131128924de882e56? My fingers are crossed that there aren't any conflicts....

**Pull request closed without merging**